src/util/input_types.ts

```diff
@@ -1,25 +1,24 @@
-import {AnySSZType, deserialize, parseType, serialize} from "@chainsafe/ssz";
-import {expandByteArray, expandInput, unexpandInput} from "./translate";
+import {Type, fromHexString, toHexString} from "@chainsafe/ssz";
 import {dumpYaml, parseYaml} from "./yaml";
 
 type InputTypeRecord = Record<string, InputType>
 
 type InputType = {
-    parse: (raw: string, type: AnySSZType) => any,
-    dump: (value: any, type: AnySSZType) => string,
+    parse: <T>(raw: string, type: Type<T>) => any,
+    dump: <T>(value: any, type: Type<T>) => string,
 }
 
 export const inputTypes: InputTypeRecord = {
     yaml: {
-        parse: (raw, type) => expandInput(parseYaml(raw), parseType(type), false),
-        dump: (value, type) => dumpYaml(unexpandInput(value, parseType(type), false)),
+        parse: (raw, type) => type.fromJson(parseYaml(raw)),
+        dump: (value, type) => dumpYaml(type.toJson(typeof value === 'number' ? value.toString() : value)),
```

Comment:
What is the `typeof` check for?

Reply:
that's when we had talked about trying to convert numbers to strings for this part